addons/account/models/account_payment.py

```diff
@@ -880,6 +879,17 @@ def write(self, vals):
         return super().write(vals)
 
     def unlink(self):
+        move_to_unlink = self.env['account.move']
+        for payment in self:
+            if not payment.move_id:
+                continue
+
+            if payment.move_id.state != 'draft':
+                payment.move_id.button_draft()
+            move_to_unlink += payment.move_id
+
+        move_to_unlink.unlink()
```

Comment:
Will this be the same?
```suggestion
        self.move_id.filtered(lambda m: m.state != 'draft').button_draft()
        self.move_id.unlink()
```

Reply:
Yeah that's something i'm wondering just now while looking at the pr ahah, i will test it :smile: